### PR TITLE
feat(flutter): Desktop support for `stream_chat_flutter`

### PR DESCRIPTION
**Pull request by GroovinChip**
## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

This PR contains all of my work on `stream_chat_flutter` to support desktop platforms. The work that needs to be done is as follows:

- [x] #919 
- [x] #920 
- [x] #921 
- [x] #923
- [x] #1008 
- [x] #926  
- [x] #927 
- [x] #930 
- [x] #931 
- [x] #924 
- [x] #925 
- [x] Extract `PlatformWidgetBuilder` to its own package
- [x] Test changes in light mode
- [x] #1013
- [x] #960
- [x] #1018 
- [x] #1032
- [ ] #1040
---
### New tasks as of 5/25/22
- [x] Pass `userAvatarBuilder` to `MessageWidgetContent`
- [x] Pass `usernameBuilder` to `MessageWidgetContent`
- [x] Fix "reply" context menu action not working in some cases
- [x] Fix Giphys not loading in `QuotedMessageWidget` when replying
- [x] Fix being able to edit someone else's message
- [x] Fix "show in chat" context menu action not working
- [x] Fix "scroll to bottom" widget not showing
- [x] Fix doubled date dividers
- [x] Fix larger than normal link preview messages
- [x] Fix larger than normal text-only messages
- [x] Fix "reply" action not working in gallery
- [x] Navigate gallery items using left, right arrow keys
- [x] Fix "enter" key sending an empty message
- [x] Close gallery by pressing "escape" key
- [x] Fix `StreamMessageListView` not scrolling to the bottom when new messages are sent
- [x] Fix gallery nav control colors
- [x] Revamp UI for changing an existing reaction & showing other users who've reacted
- [x] Fix videos not loading in `QuotedMessageWidget` when replying
- [x] #1199 
- [x] Fix broken macOS build - Awaiting https://github.com/alexmercerind/dart_vlc/pull/301 to be merged
---

Additionally, this PR contains massive refactoring to existing widgets that are unrelated to the above issues, specifically:
- [x] Stop using expression syntax for `Widget build()` methods
- [x] Break out a metric ton of widgets out into their own classes/files from various `Widget build()` helper methods
- [x] Move from `Navigator.whatever` syntax to `Navigator.of(context).whatever` syntax
- [x] Mark any `void async` functions as `Future<void> async` that have not already been done so

To check if any of the above issues are compatible with Flutter Web, please view the issue labels.

### CODE REVIEW

**Comment by GroovinChip** [comment]
@imtoori merge conflicts with the new `desktop_beta` branch (which is based off of the `v4` branch) have been resolved. That said, there are some errors related to the `scrollable_positioned_list` package. Please run the `stream_chat_flutter` example app to see these errors.

File: packages/stream_chat_flutter/lib/src/message_list_view/message_list_view.dart [hunk 1]
```diff
@@ -1309,62 +1148,6 @@ class _StreamMessageListViewState extends State<StreamMessageListView> {
...
-
-    _getOnThreadTap();
-    super.initState();
-  }
-
-  @override
-  void didChangeDependencies() {
-    final newStreamChannel = StreamChannel.of(context);
-    _streamTheme = StreamChatTheme.of(context);
-    _userPermissions = newStreamChannel.channel.ownCapabilities;
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_list_view/message_list_view.dart:1328]: @GroovinChip

Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/message_list_view/message_list_view.dart:1328]: What specifically is missing? The `_userPermissions = newStreamChannel.channel.ownCapabilities;` line?

Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/message_list_view/message_list_view.dart:1328]: Fixed in 4247dde

**Comment by GroovinChip** [comment]
@imtoori merge conflicts have been resolved.

**Pull request merged**